client/src/app/site/common/components/user-statistics/user-statistics.component.ts

```diff
@@ -0,0 +1,221 @@
+    }
+
+    public get numberOfUniqueSpeakers(): number {
+        return this.uniqueSpeakers.length;
+    }
+
+    public get numberOfWordRequests(): number {
+        return this._numberOfWordRequests;
+    }
+
+    /**
+     * Returns an observable containing a list. This list contains objects separated by the structure-level of speakers.
+     *
+     * Those objects hold information about number of word requests and duration of word requests for a given
+     * structure-level.
+     */
+    public get statisticsByStructureLevelObservable(): Observable<SpeakersOfStructureLevel[]> {
+        return this.speakersOfStructureLevelSubject.asObservable();
+    }
+
+    /**
+     * Dedicated, if statistics are opened.
+     */
+    public get openedStatistics(): boolean {
+        return this.statisticIsOpen;
+    }
+
+    public readonly columnDefinition: PblColumnDefinition[] = [
+        {
+            prop: 'structureLevel',
+            width: 'auto',
+            label: 'Structure level'
+        },
+        {
+            prop: 'durationOfWordRequests',
+            width: 'auto',
+            label: 'Duration of word requests'
+        },
+        {
+            prop: 'numberOfWordRequests',
+            width: 'auto',
+            label: 'Number of word requests'
+        }
+    ];
+
+    public readonly filterProps: string[] = ['structureLevel'];
+
+    /**
+     * Holds information about hours, minutes and seconds for the total duration of word requests.
+     */
+    private durationWordRequest: TimeObject = null;
+
+    /**
+     * List of unique speakers.
+     */
+    private uniqueSpeakers: ViewSpeaker[] = [];
+    private speakersOfStructureLevel: Map<string, SpeakersOfStructureLevel> = new Map();
+    private _numberOfWordRequests = 0;
+    private statisticIsOpen = false;
+    private speakersOfStructureLevelSubject = new BehaviorSubject<SpeakersOfStructureLevel[]>([]);
+
+    public constructor(
+        title: Title,
+        protected translate: TranslateService,
+        matSnackBar: MatSnackBar,
+        private losRepo: ListOfSpeakersRepositoryService,
+        private cd: ChangeDetectorRef
+    ) {
+        super(title, translate, matSnackBar);
+    }
+
+    public ngOnInit(): void {
+        this.subscriptions.push(
+            this.losRepo.getViewModelListObservable().subscribe(losList => {
+                this.calcWordRequests(losList.filter(los => los.closed));
+                this.pushNextState();
+            })
+        );
+    }
+
+    /**
+     * Opens or closes statistics.
+     */
+    public changeViewOfStatistics(): void {
+        this.statisticIsOpen = !this.statisticIsOpen;
+    }
+
+    /**
+     * This iterates over a list of list-of-speakers. For each speaker it calculates the duration the speaker
+     * has spoken.
+     *
+     * @param listOfSpeakers List of list-of-speakers.
+     */
+    private calcWordRequests(listOfSpeakers: ViewListOfSpeakers[]): void {
```

Comment:
Should also be handled of the list-of-speaker repo. Basically the length of finished-speakers.